src/frida/application.py

```diff
@@ -250,9 +250,10 @@ def expand_target(target):
 
 
 class Reactor(object):
-    def __init__(self, run_until_return):
+    def __init__(self, run_until_return, on_stop):
```

Comment:
Let's not add a callback here, but instead just have the base class `ConsoleApplication` call a method that might be overridden (defaulting to doing nothing). It is after all the one calling `run()`, so it has full control.